lib/api/core/plugins/pluginsManager.js

```diff
@@ -150,22 +161,19 @@ class PluginsManager {
           initTimeout
         } = this.config.common;
 
-      if (plugin.config.threads) {
+      if (plugin.object.threadable && plugin.config.threads) {
         debug('[%s] starting worker with %d threads', pluginName, plugin.config.threads);
 
         return initWorkers(plugin, pluginName, this.kuzzle.config)
           .timeout(initTimeout)
-          .catch(err => {
-            // eslint-disable-next-line no-console
-            this.silent || console.log('Plugin', pluginName, ' errored with message: ', err.message, '. Skipping...');
-            return Bluebird.resolve();
-          });
+          .catch(err => Bluebird.reject(new PluginImplementationError(`Unable to start worker plugin "${pluginName}"; ${err.message}`)));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove curly braces and "return" from this arrow function body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3524)